<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>AlgorithmTuner</class>
 <widget class="QMainWindow" name="AlgorithmTuner">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1519</width>
    <height>1004</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="tabShape">
   <enum>QTabWidget::Rounded</enum>
  </property>
  <property name="dockNestingEnabled">
   <bool>true</bool>
  </property>
  <property name="dockOptions">
   <set>QMainWindow::AllowNestedDocks|QMainWindow::AllowTabbedDocks|QMainWindow::AnimatedDocks</set>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="3" column="0">
     <widget class="QVTKWidget" name="vtkWidget" native="true"/>
    </item>
    <item row="4" column="0">
     <widget class="QProgressBar" name="progressBar">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="maximum">
       <number>10</number>
      </property>
      <property name="value">
       <number>1</number>
      </property>
      <property name="textVisible">
       <bool>true</bool>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="invertedAppearance">
       <bool>false</bool>
      </property>
      <property name="format">
       <string>%v/%m Data points</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1519</width>
     <height>24</height>
    </rect>
   </property>
   <property name="defaultUp">
    <bool>false</bool>
   </property>
   <property name="nativeMenuBar">
    <bool>true</bool>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>Dataset</string>
    </property>
    <addaction name="actionLoadDataset"/>
    <addaction name="actionNonMaximaSupression"/>
   </widget>
   <widget class="QMenu" name="menuDockableWidget">
    <property name="title">
     <string>DockWidgets</string>
    </property>
    <addaction name="actionSettings"/>
    <addaction name="actionDatapoint"/>
    <addaction name="actionResults"/>
    <addaction name="actionDataProc"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuDockableWidget"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QDockWidget" name="algorithmSettingsDockWidget">
   <property name="windowTitle">
    <string>Settings</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_4">
    <layout class="QGridLayout" name="gridLayout_2">
     <item row="1" column="0">
      <layout class="QVBoxLayout" name="sidePanelVerticalLayout">
       <property name="topMargin">
        <number>10</number>
       </property>
       <property name="bottomMargin">
        <number>13</number>
       </property>
       <item>
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>General Settings</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QFormLayout" name="generalSettingsFormLayout">
         <property name="topMargin">
          <number>10</number>
         </property>
        </layout>
       </item>
       <item>
        <widget class="Line" name="line">
         <property name="frameShadow">
          <enum>QFrame::Plain</enum>
         </property>
         <property name="lineWidth">
          <number>5</number>
         </property>
         <property name="midLineWidth">
          <number>5</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="algorithmSettingsQLabel">
         <property name="text">
          <string>Algorithm Settings</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QTabWidget" name="settingsTabWidget">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="currentIndex">
          <number>0</number>
         </property>
         <property name="elideMode">
          <enum>Qt::ElideNone</enum>
         </property>
         <property name="tabsClosable">
          <bool>false</bool>
         </property>
         <property name="movable">
          <bool>true</bool>
         </property>
         <property name="tabBarAutoHide">
          <bool>false</bool>
         </property>
         <widget class="QWidget" name="evaluatorTab">
          <attribute name="title">
           <string>Evaluator</string>
          </attribute>
          <layout class="QGridLayout" name="gridLayout_8">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item row="0" column="0">
            <widget class="QTabWidget" name="evaluatorTabWidget"/>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="hvAlgTab">
          <attribute name="title">
           <string>HV Algorithm</string>
          </attribute>
          <layout class="QGridLayout" name="gridLayout_6">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <property name="spacing">
            <number>0</number>
           </property>
           <item row="0" column="0">
            <widget class="QTabWidget" name="hvAlgTabWidget">
             <property name="styleSheet">
              <string notr="true"/>
             </property>
             <property name="currentIndex">
              <number>-1</number>
             </property>
             <property name="elideMode">
              <enum>Qt::ElideNone</enum>
             </property>
             <property name="documentMode">
              <bool>false</bool>
             </property>
             <property name="tabsClosable">
              <bool>false</bool>
             </property>
             <property name="movable">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
       <item>
        <widget class="QFrame" name="runFrame">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>7</width>
           <height>8</height>
          </size>
         </property>
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QGridLayout" name="gridLayout_3" columnstretch="0,0,0">
          <property name="leftMargin">
           <number>4</number>
          </property>
          <property name="topMargin">
           <number>4</number>
          </property>
          <property name="rightMargin">
           <number>4</number>
          </property>
          <property name="bottomMargin">
           <number>4</number>
          </property>
          <property name="horizontalSpacing">
           <number>4</number>
          </property>
          <item row="1" column="0">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Repititions</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="1" column="1" colspan="2">
           <widget class="QSpinBox" name="repetitionSpinBox">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>1000000</number>
            </property>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QDoubleSpinBox" name="paramMinDoubleSpinBox">
            <property name="decimals">
             <number>3</number>
            </property>
            <property name="minimum">
             <double>-100000.000000000000000</double>
            </property>
            <property name="maximum">
             <double>10000.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="6" column="2">
           <widget class="QDoubleSpinBox" name="paramStepDoubleSpinBox">
            <property name="decimals">
             <number>5</number>
            </property>
            <property name="minimum">
             <double>-10000.000000000000000</double>
            </property>
            <property name="maximum">
             <double>10000.000000000000000</double>
            </property>
            <property name="value">
             <double>1.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="9" column="2">
           <widget class="QPushButton" name="tunedParamSavePushButton">
            <property name="text">
             <string>Save</string>
            </property>
           </widget>
          </item>
          <item row="9" column="1">
           <widget class="QPushButton" name="tunedParamLoadPushButton">
            <property name="text">
             <string>Load</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0" colspan="3">
           <widget class="QLabel" name="label_5">
            <property name="font">
             <font>
              <weight>75</weight>
              <bold>true</bold>
              <underline>true</underline>
             </font>
            </property>
            <property name="text">
             <string>Parameter Range Testing</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="4" column="1" colspan="2">
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>Parameter</string>
            </property>
           </widget>
          </item>
          <item row="9" column="0">
           <widget class="QCheckBox" name="tunedParamcheckBox">
            <property name="text">
             <string>Enable</string>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Algorithm</string>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <widget class="QDoubleSpinBox" name="paramMaxDoubleSpinBox">
            <property name="decimals">
             <number>3</number>
            </property>
            <property name="minimum">
             <double>-10000.000000000000000</double>
            </property>
            <property name="maximum">
             <double>10000.000000000000000</double>
            </property>
            <property name="value">
             <double>1.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="8" column="0" colspan="3">
           <widget class="QLabel" name="label_8">
            <property name="font">
             <font>
              <weight>75</weight>
              <bold>true</bold>
              <underline>true</underline>
             </font>
            </property>
            <property name="text">
             <string>Tuned Parameters</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QPushButton" name="runEnabledMethodsButton">
            <property name="text">
             <string>Run</string>
            </property>
           </widget>
          </item>
          <item row="10" column="0" colspan="3">
           <widget class="QTextEdit" name="tunedParamstextEdit">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
           </widget>
          </item>
          <item row="5" column="1" colspan="2">
           <widget class="QComboBox" name="paramComboBox"/>
          </item>
          <item row="5" column="0">
           <widget class="QComboBox" name="algComboBox">
            <item>
             <property name="text">
              <string>None</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="0" column="1" colspan="2">
           <widget class="QComboBox" name="runEnabledMethodsComboBox">
            <property name="sizeAdjustPolicy">
             <enum>QComboBox::AdjustToContentsOnFirstShow</enum>
            </property>
            <item>
             <property name="text">
              <string>Data Point</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Object</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Dataset</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="datasetDock">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>295</width>
     <height>218</height>
    </size>
   </property>
   <property name="windowTitle">
    <string>Dataset</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_5">
    <layout class="QFormLayout" name="formLayout_4">
     <item row="0" column="0">
      <widget class="QLabel" name="objectNameLabel">
       <property name="text">
        <string>Object Name</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QComboBox" name="objectNameComboBox"/>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="datapointLabel">
       <property name="text">
        <string>Datapoint #</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QSpinBox" name="datapointSpinBox"/>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="label">
       <property name="text">
        <string>OCs</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QLabel" name="ocsCountLabel">
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>GTs</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QLabel" name="gtsCountLabel">
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dataProcDock">
   <property name="floating">
    <bool>false</bool>
   </property>
   <property name="windowTitle">
    <string>Data</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QGridLayout" name="gridLayout_4">
     <item row="4" column="0">
      <layout class="QGridLayout" name="gridLayout_5">
       <property name="sizeConstraint">
        <enum>QLayout::SetDefaultConstraint</enum>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <item row="2" column="0" colspan="2">
        <widget class="QPushButton" name="dataProcSaveResultsButton">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="text">
          <string>Save Data</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0" colspan="2">
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionLoadDataset">
   <property name="text">
    <string>Load Dataset</string>
   </property>
  </action>
  <action name="actionDataSelector">
   <property name="text">
    <string>DataSelector</string>
   </property>
  </action>
  <action name="actionSettings">
   <property name="text">
    <string>Settings</string>
   </property>
  </action>
  <action name="actionDatapoint">
   <property name="text">
    <string>Datapoint</string>
   </property>
  </action>
  <action name="actionResults">
   <property name="text">
    <string>Results</string>
   </property>
  </action>
  <action name="actionDataProc">
   <property name="text">
    <string>DataProc</string>
   </property>
  </action>
  <action name="actionNonMaximaSupression">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Non-Maxima Suppresion</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QVTKWidget</class>
   <extends>QWidget</extends>
   <header location="global">QVTKWidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionSettings</sender>
   <signal>triggered()</signal>
   <receiver>algorithmSettingsDockWidget</receiver>
   <slot>show()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>1419</x>
     <y>560</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionDatapoint</sender>
   <signal>triggered()</signal>
   <receiver>datasetDock</receiver>
   <slot>show()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>121</x>
     <y>480</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionDataProc</sender>
   <signal>triggered()</signal>
   <receiver>dataProcDock</receiver>
   <slot>show()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>147</x>
     <y>940</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
